src/Lib/Managers/SelectedBoundManager.vala

```diff
@@ -310,4 +318,48 @@ public class Akira.Lib.Managers.SelectedBoundManager : Object {
 
         canvas.window.event_bus.set_focus_on_canvas ();
     }
+
+    /**
+     * Move the item based on the mouse click and drag event.
+     */
+    private void move_from_event (
+        Lib.Items.CanvasItem item, double event_x, double event_y) {
+        if (!initial_drag_item_registered) {
+            initial_drag_item_registered = true;
+            initial_drag_item_x = item.bounds.x1;
+            initial_drag_item_y = item.bounds.y1;
+        }
+
+        // Keep reset and delta values for future adjustments.
+
+        // Calculate values needed to reset to the original position
+        var reset_x = item.bounds.x1 - initial_drag_item_x;
+        var reset_y = item.bounds.y1 - initial_drag_item_y;
+
+        // Calculate the change based on the event
+        var delta_x = event_x - initial_drag_press_x;
+        var delta_y = event_y - initial_drag_press_y;
+
+        // Keep reset and delta values for future adjustments. fix_size should
+        // be called right before a transform.
+        var first_move_x = Utils.AffineTransform.fix_size(delta_x - reset_x);
+        var first_move_y = Utils.AffineTransform.fix_size(delta_y - reset_y);
+
+        Cairo.Matrix matrix;
+        item.get_transform (out matrix);
+
+        // Increment the cairo matrix coordinates so we can ignore the item's rotation.
+        matrix.x0 += first_move_x;
+        matrix.y0 += first_move_y;
+
+        item.set_transform (matrix);
+
+        // Any adjustments like snapping would be computed here, and the item would
+        // then be translated again.
+
+        // If the item is an Artboard, move the label with it.
+        if (item is Lib.Items.CanvasArtboard) {
+            ((Lib.Items.CanvasArtboard) item).label.translate (first_move_x, first_move_x);
+        }
```

Comment:
This doesn't work.
Create an Artboard and move it around, you'll see how the label is a bit wonky.

Reply:
Ah, I found the error. You're using `first_move_x` also for the Y :P